packages/mermaid/src/schemas/config.schema.yaml

```diff
@@ -1853,6 +1856,32 @@ $defs: # JSON Schema definition (maybe we should move these to a seperate file)
         type: string
         default: ''
 
+  PacketDiagramConfig:
+    title: Packet Diagram Config
+    allOf: [{ $ref: '#/$defs/BaseDiagramConfig' }]
+    description: The object containing configurations specific for packet diagrams.
+    type: object
+    unevaluatedProperties: false
+    properties:
+      rowHeight:
+        description: The height of each row in the packet diagram.
+        default: 32
+      bitWidth:
+        description: The width of each bit in the packet diagram.
+        default: 32
+      bitsPerRow:
+        description: The number of bits to display per row.
+        default: 32
+      showBits:
+        description: Toggle to display or hide bit numbers.
+        default: true
+      paddingX:
+        description: The horizontal padding between the blocks in a row.
+        default: 5
+      paddingY:
+        description: The vertical padding between the rows.
+        default: 5
```

Comment:
I'm not fimilar `config.schemal.yaml`, but do they need `type` attribute or not?

```yaml
      rowHeight:
        description: The height of each row in the packet diagram.
        type: number
        default: 32
```

Reply:
Yeah, I'm also a bit confused here. I initially had the type, but it generated the correct type from the default value for one field that didn't have type. So I tried removing type from everything, and it still worked fine. 

@aloisklink are types inferred from defaults or should we explicitly state them anyway?

Reply:
> @aloisklink are types inferred from defaults or should we explicitly state them anyway?

Officially, if there is no `type` field, any type should be allowed (if the `json-schema-to-typescript` converter is using `typeof` the `default` value instead of `any`, I believe this is technically wrong).

For documentation/TypeScript purposes, I would definitely recommend putting in an explicit type, so people know not to put in strings/arrays/objects into these types.

Btw, if you want to limit a value to just positive integers, you can write:

```yaml
type: integer
minimum: 0
```